lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java

```diff
@@ -96,19 +120,27 @@ public APIGatewayProxyResponseEvent handleRequest(
             String govukSigninJourneyId = clientSessionDetailsDto.getGovukSigninJourneyId();
             LogHelper.attachGovukSigninJourneyIdToLogs(govukSigninJourneyId);
 
-            List<String> credentials = userIdentityService.getUserIssuedCredentials(userId);
+            List<SignedJWT> credentials =
+                    gpg45ProfileEvaluator.parseCredentials(
+                            userIdentityService.getUserIssuedCredentials(userId));
 
             Optional<JourneyResponse> contraIndicatorErrorJourneyResponse =
                     gpg45ProfileEvaluator.getJourneyResponseForStoredCis(clientSessionDetailsDto);
             if (contraIndicatorErrorJourneyResponse.isEmpty()) {
-                boolean credentialsSatisfyProfile =
-                        gpg45ProfileEvaluator.credentialsSatisfyAnyProfile(
-                                gpg45ProfileEvaluator.parseGpg45ScoresFromCredentials(credentials),
-                                ACCEPTED_PROFILES);
+                Gpg45Scores gpg45Scores = gpg45ProfileEvaluator.buildScore(credentials);
+                Optional<Gpg45Profile> matchedProfile =
+                        gpg45ProfileEvaluator.getFirstMatchingProfile(
+                                gpg45Scores, ACCEPTED_PROFILES);
                 JourneyResponse journeyResponse;
                 var message = new MapMessage();
 
-                if (credentialsSatisfyProfile) {
+                if (matchedProfile.isPresent()) {
+                    auditService.sendAuditEvent(
+                            buildProfileMatchedAuditEvent(
+                                    ipvSessionItem,
+                                    matchedProfile.get(),
+                                    gpg45Scores,
+                                    credentials));
                     ipvSessionItem.setVot(VOT_P2);
                     journeyResponse = JOURNEY_END;
                     message.with("lambdaResult", "A GPG45 profile has been met")
```

Comment:
Minor thing, but is it worth adding the profile that we've matched into this log message?

Reply:
We already do this in the `Gpg45ProfileEvaluator` - https://github.com/alphagov/di-ipv-core-back/pull/537/files#diff-f61afdbba01ff0a1ecb6f163d3a5e438472e54882b62b50988e02fcd12006d44R95

Reply:
Ah yeah true 👍